client/v3/naming/resolver/resolver.go

```diff
@@ -17,9 +18,15 @@ type builder struct {
 }
 
 func (b builder) Build(target gresolver.Target, cc gresolver.ClientConn, opts gresolver.BuildOptions) (gresolver.Resolver, error) {
```

Comment:
Is this backported from anywhere? Will try to find it myself, but please separate commits that backport changes as we should not diverge from main branch.

Reply:
Found https://github.com/etcd-io/etcd/pull/14922